File: discord_bot.js
const { Client, GatewayIntentBits } = require('discord.js');

// Créez une nouvelle instance de client
// Les "Intents" définissent les événements que votre bot recevra.
const client = new Client({
    intents: [
        GatewayIntentBits.Guilds,
        GatewayIntentBits.GuildMessages,
        GatewayIntentBits.MessageContent, // Nécessaire pour lire le contenu des messages
    ]
});

// Remplacez ceci par le token de votre bot
const token = 'VOTRE_TOKEN_DISCORD_ICI';

// Quand le client est prêt, ce code s'exécute (une seule fois)
client.once('ready', () => {
    console.log(`Prêt ! Connecté en tant que ${client.user.tag}`);
});

// Écoute les messages
client.on('messageCreate', message => {
    // Ignore les messages des autres bots pour éviter les boucles
    if (message.author.bot) return;

    // Vérifie si le message est la commande !ping
    if (message.content === '!ping') {
        // Répond "Pong!" avec la latence du message
        const latency = Date.now() - message.createdTimestamp;
        message.reply(`Pong ! Latence : ${latency}ms.`);
    }
});

// Connectez-vous à Discord avec le token de votre client
client.login(token);

File: discord_bot.py

import discord
from discord.ext import commands

# Définit les autorisations (intents) dont le bot a besoin.
# Pour lire les messages, '''message_content''' est nécessaire.
intents = discord.Intents.default()
intents.message_content = True

# Crée une instance de bot avec le préfixe de commande "!" et les intents.
bot = commands.Bot(command_prefix='!', intents=intents)

# Événement qui se déclenche lorsque le bot est prêt et connecté.
@bot.event
async def on_ready():
    print(f'Connecté en tant que {bot.user.name}!')

# Définit la commande !ping.
@bot.command()
async def ping(ctx):
    """Répond Pong! à la commande !ping."""
    await ctx.send('Pong!')

# Lance le bot avec votre token.
# Remplacez 'VOTRE_TOKEN_ICI' par le vrai token de votre bot.
bot.run('VOTRE_TOKEN_ICI')
